### PR TITLE
Upgrade vllm

### PR DESCRIPTION
**Pull request by Jackmin801**
> [!NOTE]
> Upgrades vLLM to 0.12 with server/router migration and config updates, bumps PyTorch to 2.9, adds optional flash-attn extra, and switches CI/Docker/install to `uv sync --all-extras`.
> 
> - **Inference/server**:
>   - Migrate to vLLM `0.12.0` APIs: use built-in `router`, `engine_client`, and `run_api_server_worker_proc`; register custom endpoints (`/update_weights`, `/reload_weights`, `/init_broadcaster`).
>   - Update Chat Completions with tokens flow to new tokenizer access, logits processors validation, prompt handling, and DP-rank propagation; refine logging/streaming paths; keep LoRA in-place monkey-patch.
>   - Filesystem worker now accesses model as `model_runner.model.runnable`.
> - **Config**:
>   - Add `max_cpu_loras` (default `100`) and map to vLLM; change `seed` to required `int` default `0`; minor field mapping updates.
> - **Build/CI/docs**:
>   - Use `uv sync --all-extras` in CPU/GPU/nightly workflows, Dockerfile, install script, and README (including FlashAttention note).
> - **Dependencies**:
>   - Bump: `vllm`→`0.12.0`, `torch`→`2.9.0` (and `torchaudio`/`torchvision`), `fastapi`/`starlette`, `pydantic`/`pydantic-core`, `triton`, plus related packages.
>   - Introduce optional `flash-attn` extra (prebuilt wheel) and lockfile updates; remove direct `flash-attn` from core deps.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit fc84210ee081254492d492553e991e604a6c6884. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/prime_rl/inference/config.py [hunk 1]
```diff
@@ -146,11 +146,11 @@ class InferenceConfig(BaseSettings):
     ] = 0.9
 
     seed: Annotated[
-        int | None,
+        int,
         Field(
             description="Seed the inference components. If None, no seeding is used. Passed to vLLM as `--seed`",
```
Comment (Jackmin801) [inline src/prime_rl/inference/config.py]:
```suggestion
            description="Seed the inference components. Passed to vLLM as `--seed`",
```

File: pyproject.toml [hunk 2]
```diff
@@ -38,7 +38,7 @@ dependencies = [
     "verifiers>=0.1.8",
     "prime-evals>=0.1.5",
     "prime-sandboxes>=0.2.5",
-    "flash-attn>=2.8.3",
+    "flash-attn @ https://github.com/mjun0812/flash-attention-prebuild-wheels/releases/download/v0.6.8/flash_attn-2.8.3+cu128torch2.9-cp312-cp312-linux_x86_64.whl",
```
Comment (samsja) [inline pyproject.toml]: you should use source here

Comment (samsja) [reply pyproject.toml]: and lets make it optional as its quite involved

**Review by Jackmin801 — COMMENTED** [review]
yee looks good. if pass tests shud be fine to merge

File: src/prime_rl/inference/config.py [hunk 3]
```diff
@@ -161,11 +171,11 @@ class InferenceConfig(BaseSettings):
...
 
     seed: Annotated[
-        int | None,
+        int,
         Field(
-            description="Seed the inference components. If None, no seeding is used. Passed to vLLM as `--seed`",
+            description="Seed the inference components. Passed to vLLM as `--seed`",
         ),
-    ] = None
+    ] = 0
```
[the remark this answers is not in the document]
Comment (Jackmin801) [reply src/prime_rl/inference/config.py:179]: eh sure bro

**Review by samsja — APPROVED** [review]

**Review by samsja — APPROVED** [review]
lgtm
